codes/c/chapter_array_and_linkedlist/array.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * File: array.c
+ * Created Time: 2022-12-20
+ * Author: MolDuM (moldum@163.com)
+ */
+
+#include "../include/include.h"
+
+/* 随机返回一个数组元素 */
+int randomAccess(int* nums, int size) {
+    // 在区间 [0, size) 中随机抽取一个数字
+    int randomIndex = rand() % size;
+    // 获取并返回随机元素
+    int randomNum = nums[randomIndex];
+    return randomNum;
+}
+
+/* 扩展数组长度 */
+int* extend(int* nums, int size, int enlarge) {
+    // 初始化一个扩展长度后的数组
+    int* res = (int *)malloc(sizeof(int) * (size + enlarge));

```

Comment:
Sorry, I'm not familiar with C. But how about initializing the array in this way:

```c
int arr[size + enlarge];
int *res = arr;
```

Reply:
I think it is not a good idea in C.
The first line defines a local variable array in the function, which means the address would be assigned to arr, and it would be freed when return. In another word, the numbers in arr would be lost after the function finished. Of course we can define a new array outside the function, but it would reduce the transportability.
Besides, the first line is not recommended in C because the length of array should be a constant value instead of a variable. There would be a reminder from IDE (but no warning and error) on my VSCode.
This is a common problem in C. In my experience, I would use malloc to define an unknown length array and use realloc to change the length. But it is not the classical way to define an array. I think I need to have a look at the official document and update it later.

Reply:
Good explanation. 👍 

Actually, I think the array in C/C++ is too complex to use. In practice, We'd like to use `vector` as an array in C++ in 100% conditions.